config.cpp

```diff
@@ -135,6 +135,44 @@ static std::filesystem::path normalizePath(const std::filesystem::path path)
     return result;
 }
 
+std::string Config::matchFilenameFromCompileCommand()
+{
+    // Read compile_commands file
+    std::ifstream ifs(m_projectFilePath);
```

Comment:
please check that the file extension is ".json". if the project file extension is for instance ".cppcheck" then I think we should just leave m_filename as-is, it's not an error.